feat: [CDS-84377]: Add examples of using output variables within step group (#6154)
harness/developer-hub: docs/continuous-delivery/x-platform-cd-features/cd-steps/step-groups.md

You can refer step group variables outside the step group using the expression, `<+pipeline.stages.[stage Id].spec.execution.steps.[step group id].variables.[variable name]>`.

### Output variables

> You can refer to output variables using expression variable expression which requires the step group name to be present, `<+execution.steps.<stepGroupID>.steps.stepID.outputVariables.[variable name]>`

Reviewer: I'm kinda confused on what expression variable expression means?

> You can refer to output variables using expression variable in expression which requires the step group name to be present, `<+execution.steps.<stepGroupID>.steps.stepID.outputVariables.[variable name]>`

Reviewer 2: ```suggestion
You can reference step group output variables using expressions using the step group name. Here's a sample expression:
`<+execution.steps.<stepGroupID>.steps.stepID.outputVariables.[variable name]>`.
```

> If you are accessing the output variable in a steo within the same step group you can use a shorter notation, `<+stepGroup.steps.step1Identifier.output.outputVariables.[variable name]>`

Reviewer: ```suggestion
If you are accessing the output variable in a step within the same step group you can use a shorter notation, `<+stepGroup.steps.step1Identifier.output.outputVariables.[variable name]>`.
```

> If your step group is configured as loop, there is an index added at the end of identifier to identify the unique loop instance you can refer the output variable in this case as, `<+execution.steps.<stepGroupID_[index]>.steps.stepID.outputVariables.[variable name]>`

Reviewer: ```suggestion
If your step group is configured as a loop, there is an index added at the end of the identifier to identify the unique loop instance you can use to refer to the output variable. In this case: `<+execution.steps.<stepGroupID_[index]>.steps.stepID.outputVariables.[variable name]>`
```

> If your step group is configured as a loop, there is an index added at the end of the identifier to identify the unique loop instance you can use to refer to the output variable. In this case: `<+execution.steps.<stepGroupID_[index]>.steps.stepID.outputVariables.[variable name]>`

Reviewer 2: ```suggestion
If your step group is configured as a loop, there is an index added at the end of the identifier to identify the unique loop instance. You can use this unique loop instance to reference the output variables. For example, `<+execution.steps.<stepGroupID_[index]>.steps.stepID.outputVariables.[variable name]>`.
```
